usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -114,24 +114,65 @@ def search_keyword_id_list_all(keyword):
     from timing out.
     """
     index_name = '{}-'.format(TRANSACTIONS_INDEX_ROOT.replace('_', ''))+'*'
-
-    query = {
-        "query": {
-            "query_string": {
-                "query": keyword
-            }
-        }, "size": DOWNLOAD_QUERY_SIZE}
+    query = {"query": {"query_string": {"query": keyword}},
+             "aggs": {
+                    "results": {
+                     "terms": {"field": "transaction_id", "size": DOWNLOAD_QUERY_SIZE}
+                    }
+                }, "size": 0}
     try:
         responses = CLIENT.search(index=index_name, body=query, timeout='3m')
     except Exception:
         logging.exception("There was an error connecting to the ElasticSearch instance.")
         return None
-    transaction_id_list = []
     try:
-        for response in (responses['hits']['hits']):
-            list_item = (response['_source']['transaction_id'])
-            transaction_id_list.append(list_item)
-        return transaction_id_list
+        responses = responses["aggregations"]['results']
+        return [floor(response['key']) for response in responses['buckets']]
     except Exception:
         logging.exception("There was an error parsing the transaction ID's")
         return None
+
+
+def get_sum_aggregation_results(keyword):
+    """
+    Size has to be zero here because you only want the aggregations
+    """
+    index_name = '{}-'.format(TRANSACTIONS_INDEX_ROOT.replace('_', ''))+'*'
+    query = {"query": {"query_string": {"query": keyword}},
+             "aggs": {"transaction_sum": {"sum": {"field": "transaction_amount"}}}}
+    try:
+        response = CLIENT.search(index=index_name, body=query)
+        return response['aggregations']
+    except Exception:
+        logging.exception("There was an error connecting to the ElasticSearch instance.")
+        return None
+
+
+def spending_by_transaction_sum(filters):
+    keyword = filters['keyword']
+    return get_sum_aggregation_results(keyword)
+
+
+def get_transaction_ids(keyword, size=500):
+    '''returns a generator that
```

Comment:
we should keep the docstrings for functions consistent with using `"""` instead of `'''`